pkg/storage/snapshot/snapshot.go

```diff
@@ -728,21 +721,32 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		} else {
 			updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Source does not exist"))
 		}
-		updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionFalse, "Not ready"))
+
 		if vmSnapshotDeleting(vmSnapshotCpy) {
 			vmSnapshotCpy.Status.Phase = snapshotv1.Deleting
```

Comment:
Is there a reason to change the phase and condition here(only if not failed/succeeded) instead of under the other `if vmSnapshotDeleting(vmSnapshotCpy)` line 741?

Reply:
Yeah I wanted to make sure the "progressing" condition is set within the scope of the `if elseif else` here https://github.com/kubevirt/kubevirt/blob/cd2b6e96b441537386aef4f9a6aca49740386634/pkg/storage/snapshot/snapshot.go#L696-L729

Reply:
Because Succeeded/Failed are terminal.  Anyway, deleting a snapshot in a terminal phase is essentially a no op and should happen immedietly.  Deleting an in progress snapshot takes time (unfreeze/unlock)

Reply:
so I still dont see a reason to add this `if` inside the progress phase. its adds code complexity for no reason as far as I see. simply do one if that handles everything connected to deleteing

Reply:
The reason is to isolate setting the phase and progressing condition to a specific block of code.

Also, if moved to the other if block have to check to make sure not already in terminal state (succeeded, failed) before setting phase/condition.  That is just as much if not more complexity.  Remember one of the major goals here is to make some phases terminal

Reply:
Since you mentioned that once completed then deleting is basically `no op` then there is not reason why not to change the terminal phases to deleting since it wont actually change anything according to you..

Reply:
But whatever I will not insist